### PR TITLE
Fix missing GUNC overwriting #514 

### LINKED ISSUE #502: GUNC output overwriting itself
### Description of the bug

Gunc outputs are being overwritten is the result directory: the output file name of gunc is `GUNC.{DB version}maxCSS_level.tsv` and `{PREFIX}.all_levels.tsv`. (see [GUNC doc](https://grp-bork.embl-community.io/gunc/output.html#output-files))

Because GUNC is run per bin in nf-core/mag, but the results are saved per `[assembler]-[binner]-[sample]` (See code [here](https://github.com/maxibor/mag/blob/49474625c1b46eb5ab16a6a7dc00e38e6cb7e0cc/conf/modules.config#L422-L423)) (and should probably be `[assembler]-[binner]-[domain]-[refinement]-[sample]` according to #489 ) and are therefore being overwritten.

Note that the gunc summary is not affected, and the `gunc_summary.tsv` file is fine.

### Command used and terminal output

_No response_

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by jfy133**
> Wait until https://github.com/nf-core/mag/pull/511 is merged! Then test before requesting review!

closes #502 to stop GUNC overwriting otuput directories when running with domain classification

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: docs/output.md [hunk 1]
```diff
@@ -557,13 +557,13 @@ If the parameter `--save_checkm_reference` is set, additionally the used the Che
 <summary>Output files</summary>
 
 - `GenomeBinning/QC/gunc_summary.tsv`
-- `GenomeBinning/QC/gunc_checkm_summary.tsv`
+- `GenomeBinning/QC/gunc_checkm_GUNCmmary.tsv`
```
Comment (jfy133) [inline docs/output.md:560]:
```suggestion
- `GenomeBinning/QC/gunc_checkm_GUNCsummmary.tsv`
```

**Comment by jfy133** [comment]
I confirm this works:

image

nextflow run ../main.nf -profile singularity,test --outdir ./results --skip_concoct false --skip_spadeshybrid --skip_prokka --skip_prodigal --run_gunc --bin_domain_classification

**Review by prototaxites — APPROVED** [review]
LGTM 👍

**Review by CarsonJM — APPROVED** [review]
Looks good to me!

**Pull request merged**